packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx

```diff
@@ -1,27 +1,119 @@
-import React, { FormEvent, useState } from 'react'
+import React, { FormEvent, useEffect, useMemo, useRef, useState } from 'react'
 
 import { Button } from '@audius/stems'
+import base64url from 'base64url'
 import * as queryString from 'query-string'
 import { useSelector } from 'react-redux'
 import { useLocation } from 'react-router-dom'
 
 import { getAccountUser } from 'common/store/account/selectors'
 import Input from 'components/data-entry/Input'
-import { MetaMaskOption } from 'pages/sign-on/components/desktop/MetaMaskOption'
 import AudiusBackend from 'services/AudiusBackend'
 
 import styles from '../styles/OAuthLoginPage.module.css'
 
 export const OAuthLoginPage = () => {
-  const { search } = useLocation()
-  const { scope, api_key, state, redirect } = queryString.parse(search)
+  const { search, hash } = useLocation()
+  const isRedirectValidRef = useRef<undefined | boolean>(undefined)
+  const { scope, state, redirect_uri } = queryString.parse(search)
+  const parsedRedirectUri = useMemo(() => {
+    if (redirect_uri && typeof redirect_uri === 'string') {
+      let res: URL
+      try {
+        res = new URL(redirect_uri)
```

Comment:
Can we just return here?
```suggestion
        return new URL(redirect_uri)
```